ocaml-lsp-server/src/workspace_symbol.ml

```diff
@@ -0,0 +1,57 @@
+open Import
+
+let kind (kind : IndexTypes.kind) : SymbolKind.t =
+  match kind with
+  | Type -> String
+  | Value -> Function
+  | Exception -> Constructor
+  | OpenType -> Module
+  | Field _
+  | Variant _ ->
+    Constructor
+  | Method _ -> Method
+  | Module
+  | ModuleType ->
+    Module
+  | Class
+  | ClassType ->
+    Class
+  | Keyword -> Key
+
+let to_loc (loc : Warnings.loc) : Loc.t =
+  { loc_start = loc.loc_start
+  ; loc_end = loc.loc_end
+  ; loc_ghost = loc.loc_ghost
+  }
+
+let symbol (info : IndexTypes.info) =
+  let loc_impl = Lazy.force_val info.loc_impl |> to_loc in
+  let loc =
+    if Loc.is_none loc_impl then
+      Lazy.force_val info.loc_sig |> to_loc
+    else
+      loc_impl
+  in
+  let uri = loc.loc_start.pos_fname in
+  let range = Range.of_loc loc in
+  let location : Location.t = { uri; range } in
+  let name = LibIndex.Print.path info in
+  let kind = kind info.kind in
+  SymbolInformation.create ~name ~kind ~location ()
+
+let run ({ query } : WorkspaceSymbolParams.t) =
+  let _, build_dir = IndexMisc.project_root () in
```

Comment:
Any better way to get the `_build` directory path? From what I understand, this function walks up the tree starting from the current dir until it finds the build directory.

Also, what if we have multiple workspaces? I haven't tested that yet and wondering if we should do anything special?

Reply:
> Any better way to get the _build directory path? From what I understand, this function walks up the tree starting from the current dir until it finds the build directory.

This is ok for now. Eventually we'll have better integration with dune.

> Also, what if we have multiple workspaces? I haven't tested that yet and wondering if we should do anything special?

The user will need to run a server per workspace then. The model we should stick to is 1 lsp server per dune workspace.